### PR TITLE
Enhanced RDoc for Integer

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #allbits?
- #anybits?
- #nobits?
- #succ
- #pred
- #chr`
- #to_s
- #+
- #-

### CODE REVIEW

**Comment by BurdetteLamar** [comment]
@jeremyevans, you have done yeoman's work for these two years now, reviewing RDoc changes (for which many thanks!).  Would you like me to request some reviews by others?

**Comment by jeremyevans** [comment]
I should have time to review this tomorrow.  I was at RubyConf the last two days and didn't have time during the conference.

File: numeric.c [hunk 1]
```diff
@@ -3338,9 +3338,30 @@ rb_int_even_p(VALUE num)
...
  *  call-seq:
- *     int.allbits?(mask)  ->  true or false
+ *    allbits?(mask) -> true or false
+ *
+ *  Returns +true+ if all bits that are set (=1) in +mask+
+ *  are also set in +self+; returns +false+ otherwise.
+ *
+ *  Example values:
+ *
+ *    0b1010101  self
```
Comment (jeremyevans) [inline numeric.c:3348]: I would cut this example and keep the other two.

File: numeric.c [hunk 2]
```diff
@@ -3352,9 +3373,30 @@ int_allbits_p(VALUE num, VALUE mask)
...
+ *  is also set in +self+; returns +false+ otherwise.
+ *
+ *  Example values:
+ *
+ *    0b10000010  self
+ *    0b11111111  mask
+ *    0b10000010  self & mask
+ *          true  self.anybits?(mask)
+ *
+ *    0b10000010  self
```
Comment (jeremyevans) [inline numeric.c:3388]: I would cut this example and leave the other two.

File: numeric.c [hunk 3]
```diff
@@ -3366,9 +3408,30 @@ int_anybits_p(VALUE num, VALUE mask)
...
  *  call-seq:
- *     int.nobits?(mask)  ->  true or false
+ *    nobits?(mask) -> true or false
+ *
+ *  Returns +true+ if no bit that is set (=1) in +mask+
+ *  is also set in +self+; returns +false+ otherwise.
+ *
+ *  Example values:
+ *
+ *    0b00000000  self
```
Comment (jeremyevans) [inline numeric.c:3418]: I would cut this example and leave the other two.

**Review by jeremyevans — APPROVED** [review]
This looks good.  I would cut the examples for `*bits?` to two example sections in each, but I'll leave up to you.

File: numeric.c (hunk 1 above)
Comment (BurdetteLamar) [reply numeric.c:3348]: Done.

File: numeric.c (hunk 3 above)
Comment (BurdetteLamar) [reply numeric.c]: Done.

File: numeric.c (hunk 2 above)
Comment (BurdetteLamar) [reply numeric.c]: Done.

**Review by jeremyevans — APPROVED** [review]
Looks good!

**Comment by ioquatix** [comment]
Unfortunately this has broken the master branch because of ` *     255.chr(Encoding::UTF_8) # => "ÿ"` can we please fix? Thanks! :)

**Comment by duerst** [comment]
If the issue with the broken master branch is the "ÿ", then we better fix the underlying problem.

**Comment by mame** [comment]
I have tentatively reverted the change due to the CI failure. Please do not merge such a commit.

**Comment by BurdetteLamar** [comment]
> 
> 
> Unfortunately this has broken the master branch because of ` * 255.chr(Encoding::UTF_8) # => "ÿ"` can we please fix? Thanks! :)

I'd like to preserve the offending character, because it's in the actual output from irb.

I don't see a similar check for .rb files, so am considering moving the doc for #chr from numeric.c to numeric.rb, with added  `#  Document-method: Integer#chr
`.  Does that make sense?

**Comment by duerst** [comment]
> I don't see a similar check for .rb files, so am considering moving the doc for #chr from numeric.c to numeric.rb, with added `# Document-method: Integer#chr `. Does that make sense?

If that help, then please give it a try.

**Comment by BurdetteLamar** [comment]
@duerst, worked this out with @jeremyevans, so that the doc remains in .c.

**Pull request merged**